Packs/Base/Scripts/DBotFindSimilarIncidents/DBotFindSimilarIncidents.py

```diff
@@ -0,0 +1,854 @@
+        )
+
+    if confidence:
+        similar_incidents = similar_incidents[similar_incidents.final_score >= confidence]
+    if show_distance == 'False':
+        col_to_remove = ['similarity %s' % field for field in fields_used]
+        similar_incidents.drop(col_to_remove, axis=1, inplace=True)
+    if include_indicators_similarity == "True":
+        similar_incidents = similar_incidents.sort_values(by=ORDER_SCORE_WITH_INDICATORS, ascending=False)
+    else:
+        similar_incidents = similar_incidents.sort_values(by=ORDER_SCORE_NO_INDICATORS, ascending=False)
+
+    return similar_incidents.head(max_incidents)
+
+
+def get_incidents_to_predict(incident_id: str, populate_fields: List[str]):
+    """
+    Get incident acording to incident id
+    :param incident_id:
+    :param populate_fields:
+    :return: Get incident acording to incident id
+    """
+    res = demisto.executeCommand('GetIncidentsByQuery', {
+        'query': "id:(%s)" % incident_id,
+        'populateFields': ' , '.join(populate_fields)
+    })
+    if is_error(res):
+        return_error(res)
+    if not json.loads(res[0]['Contents']):
+        return None
+    else:
+        incident = json.loads(res[0]['Contents'])
+        return incident[0]
+
+
+def get_all_incidents_for_time_window_and_exact_match(exact_match_fields: List[str], populate_fields: List[str],
+                                                      incident: Dict, from_date: str, to_date: str,
+                                                      query_sup: str, limit: int):
+    """
+    Get incidents for a time window and exact match for somes fields
+    :param exact_match_fields: List of field for exact match
+    :param populate_fields: List of field to populate
+    :param incident: json representing the current incident
+    :param from_date: from_date
+    :param to_date: to_date
+    :param query_sup: additional query
+    :param limit: limit of how many incidents we want to query
+    :return:
+    """
+    msg = ""
+    exact_match_fields_list = []
+    for exact_match_field in exact_match_fields:
+        if exact_match_field not in incident.keys():
+            msg += "%s \n" % MESSAGE_NO_FIELD % exact_match_field
+        else:
+            exact_match_fields_list.append('%s: "%s"' % (exact_match_field, incident[exact_match_field]))
+    query = " AND ".join(exact_match_fields_list)
+    query += " AND -id:%s " % incident['id']
+    if query_sup:
+        query += " %s" % query_sup
+
+    res = demisto.executeCommand('GetIncidentsByQuery', {
+        'query': query,
+        'populateFields': ' , '.join(populate_fields),
+        'fromDate': from_date,
+        'toDate': to_date,
+        'limit': limit
+    })
+    if is_error(res):
+        return_error(res)
+    incidents = json.loads(res[0]['Contents'])
+    if len(incidents) == 0:
+        msg += "%s \n" % MESSAGE_NO_INCIDENT_FETCHED
+        return None, msg
+    if len(incidents) == limit:
```

Comment:
seems like  `== `need to be replaced with `>=` and no truncation is done effectively

Reply:
We want to truncated the query to limit because it will overload the server. 
So the max will always be == limit

Reply:
my bad, didn't notice it's passed as an argument to `GetIncidentsByQuery`